src/main/java/app/linguistai/bmvp/service/AccountService.java

```diff
@@ -99,16 +103,29 @@ public void loginWithValidToken(String email) throws Exception {
         }
     }
   
-    public Page<User> searchUser(QUserSearch userSearch, String userEmail) throws Exception {
+    public PageImpl<RUserSearch> searchUser(QUserSearch userSearch, String userEmail) throws Exception {
         try {
+            System.out.println(userSearch.getPage());
+            User dbUser = accountRepository.findUserByEmail(userEmail).orElseThrow(() -> new LoginException());
+
             // Create a page request using the request body
             PageRequest pageable = PageRequest.of(userSearch.getPage(), userSearch.getSize());
 
-            Page<User> users = accountRepository.findByUsernameStartingWithAndEmailNot(userSearch.getUsername(), userEmail, pageable);
+            Page<Object[]> users = accountRepository.findByUsernameStartingWithAndWithFriendshipStatusAndEmailNot(userSearch.getUsername(), dbUser.getId(), pageable);
```

Comment:
Please add a comment specifying the purpose of "EmailNot" as we discussed, perhaps in a single line like "does not include the searching user" etc.

Reply:
okay